tests/Engagement_unit_test.py

```diff
@@ -0,0 +1,141 @@
+from selenium import webdriver
+from selenium.webdriver.support.ui import Select
+import unittest
+import re
+import sys
+import os
+
+
+class EngagementTest(unittest.TestCase):
+    def setUp(self):
+        self.driver = webdriver.Chrome('chromedriver')
+        self.driver.implicitly_wait(30)
+        self.base_url = "http://localhost:8000/"
+        self.verificationErrors = []
+        self.accept_next_alert = True
+
+    def login_page(self):
+        driver = self.driver
+        driver.get(self.base_url + "login")
+        driver.find_element_by_id("id_username").clear()
+        driver.find_element_by_id("id_username").send_keys(os.environ['DD_ADMIN_USER'])
+        driver.find_element_by_id("id_password").clear()
+        driver.find_element_by_id("id_password").send_keys(os.environ['DD_ADMIN_PASSWORD'])
+        driver.find_element_by_css_selector("button.btn.btn-success").click()
+        return driver
+
+    def create_new_product_to_test_engagement(self):
```

Comment:
ProperSam is working on a way to import this test from the Product test suite to reduce the amount of copy/paste code. coordinate with him to find a way to do that in a manner that is compatible with python3. Check his PRs for reference on what has already been attempted.

Reply:
okk.

Reply:
oh!.
about that, I have already created a line of code using try and except to decide between executing python 2 or python 3 code for importing your create product from another module.
of course, you would have to tweak it to use create product code from your own product_type_test.py file

 check this commit https://github.com/DefectDojo/django-DefectDojo/pull/1166/commits/ab7b88ecfcd2fad5c147206e7b4c5b1023a43f41

Reply:
don't forget to modify ur build Test suit function  at the button..

scroll down in this earlier commit to see how I did mine..

https://github.com/DefectDojo/django-DefectDojo/pull/1167/commits/0516f8c66553c93ea68fbb229148c1e00029c57a

Reply:
thanks, @propersam but I have not created any additional product in product type test so I can't just import add product function from another file (bcs i dont have :) ) the only way is we have to wait for either your pr will be merged or mine and then we can import add product function.

Reply:
Product unit test has been merged. Probably be easiest to just copy/paste from the Endpoint unit test to import the tests. Best of luck!